backend/dataall/modules/dataset_sharing/services/share_processors/lf_process_same_account_share.py

```diff
@@ -95,10 +95,10 @@ def process_approved_shares(self) -> bool:
                 shared_item_SM.update_state_single_item(self.session, share_item, new_state)
 
             except Exception as e:
-                self.handle_share_failure(table, share_item, e)
                 new_state = shared_item_SM.run_transition(ShareItemActions.Failure.value)
                 shared_item_SM.update_state_single_item(self.session, share_item, new_state)
                 success = False
+                self.handle_share_failure(table, share_item, e)
```

Comment:
I would revert these swaps.. because if transition fails the alarm will never be sent because it throws an exception.. If we leave it as it was alarm will be sent or fail quietly with logging regardless whether the share state transition fails or not...

Reply:
Done!